src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -179,15 +180,24 @@ def wrap(*args, **kwargs):
                     query = getattr(e, "query", None)
                     tb = sys.exc_info()[2]
                     assert e.msg is not None
-
-                    # extract df_ast_id, stmt_cache from args
-                    df_ast_id, stmt_cache = None, None
+                    # extract top_plan, df_ast_id, stmt_cache from args
+                    top_plan, df_ast_id, stmt_cache = None, None, None
                     for arg in args:
                         if isinstance(arg, SnowflakePlan) and arg.df_ast_ids:
+                            top_plan = arg
                             df_ast_id = arg.df_ast_ids[-1]
                             stmt_cache = arg.session._ast_batch._bind_stmt_cache
                             break
-                    df_transform_debug_trace = None
+                    error_source_context = ""
+                    if (
+                        "SQL compilation error:" in e.msg
+                        and "error line" in e.msg
+                        and top_plan is not None
+                    ):
+                        error_source_context = get_python_source_from_sql_error(
+                            top_plan, e.msg
```

Comment:
lets move this part into the try catch below so we log any errors using `_logger.info`